packages/dds/tree/src/test/simple-tree/toMapTree.spec.ts

```diff
@@ -254,25 +316,133 @@ describe("toMapTree", () => {
 			assert.deepEqual(actual, expected);
 		});
 
-		it("Empty array", () => {
+		it("Recursive array", () => {
 			const schemaFactory = new SchemaFactory("test");
-			const schema = schemaFactory.array("array", schemaFactory.number);
+			const schema = schemaFactory.arrayRecursive("array", [
+				schemaFactory.number,
+				() => schema,
+			]);
 
-			const tree: number[] = [];
+			const tree = [42, [1, 2], 37];
 
 			const actual = nodeDataToMapTree(tree, [schema]);
 
 			const expected: MapTree = {
 				type: brand("test.array"),
-				fields: new Map<FieldKey, MapTree[]>(),
+				fields: new Map<FieldKey, MapTree[]>([
+					[
+						EmptyKey,
+						[
+							{
+								type: leaf.number.name,
+								value: 42,
+								fields: new Map(),
+							},
+							{
+								type: brand("test.array"),
+								fields: new Map<FieldKey, MapTree[]>([
+									[
+										EmptyKey,
+										[
+											{
+												type: leaf.number.name,
+												value: 1,
+												fields: new Map(),
+											},
+											{
+												type: leaf.number.name,
+												value: 2,
+												fields: new Map(),
+											},
+										],
+									],
+								]),
+							},
+							{
+								type: leaf.number.name,
+								value: 37,
+								fields: new Map(),
+							},
+						],
+					],
+				]),
 			};
 
 			assert.deepEqual(actual, expected);
 		});
+
+		it("Throws on `undefined` entries", () => {
```

Comment:
We have some implicit conversion to null policy when null is allowed right? Might be good to test that variant of this policy, as well as make this test name more specific, like:
```suggestion
		it("Throws on `undefined` entries when null is not allowed", () => {
```

Reply:
> We have some implicit conversion to null policy when null is allowed right?
That is correct, yes.

> Might be good to test that variant of this policy
That is tested, elsewhere in the file. There is a section dedicated to implicit fallback policies.